### PR TITLE
`Tutorial groups`: Change channel prefix for tutorial-groups

### PR DESCRIPTION
**Pull request by lennart-keller**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I modified multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context

Automatically generated channels for lecture, exercises and exam have a certain prefix (`lecture-`, `exercise-`, `exam-`). This PR introduces such a prefix for tutorial-group channels.

### Description

The automatically generated channel name for tutorial groups is prefixed with `tutorgroup-`, instead of `$`. Additionally, the maximum allowed length for the name got increased to 30, which is the same length as for other channels.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Create or edit the global tutorial group configuration of the course: Make sure that "Use Artemis managed Tutorial Group Channels" is enabled
2. Create a tutorial group
3. A channel should be created in the form of `tutorgroup-title-of-tutorgroup`

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by DominikRemo — APPROVED** [review]
LGTM

File: src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupChannelManagementService.java [hunk 1]
```diff
@@ -250,7 +250,7 @@ private String determineUniqueTutorialGroupChannelName(TutorialGroup tutorialGro
         Function<TutorialGroup, String> determineInitialTutorialGroupChannelName = (TutorialGroup tg) -> {
             var cleanedTitle = tg.getTitle().replaceAll("\\s", "-").toLowerCase();
             // we use $ as prefix to make sure that the channel name is unique (users not allowed to start channel names with $)
```
Comment (basak-akan) [inline src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupChannelManagementService.java:252]: Since no longer $ sign is used as a prefix, this comment seems irrelevant

**Review by basak-akan — DISMISSED** [review]
Code looks good 👍 I have just a small comment.

**Review by reschandreas — APPROVED** [review]
Tested locally, works code also lgtm

**Review by laurenzfb — APPROVED** [review]
Demonstrated in testing session, code LGTM as well

**Review by rstief — APPROVED** [review]
Tested in Testing Session.

**Review by JohannesStoehr — APPROVED** [review]
Changes and code look good to me.
But we should definitely unite all these occurrences of `30` and other numbers into some constants in the future!

**Review by florian-glombik — APPROVED** [review]
Tested on ts1, and the code looks fine, as the magic numbers seem to be present at multiple places right now and should probably be replaced within one dedicated PR

**Review by reschandreas — APPROVED** [review]
reapprove
